source/base/mpi_remote_point_evaluation.cc

```diff
@@ -197,28 +197,69 @@ namespace Utilities
           if (dummy != std::get<0>(i))
             {
               dummy = std::get<0>(i);
-              cell_data.cells.emplace_back(dummy);
-              cell_data.reference_point_ptrs.emplace_back(
-                cell_data.reference_point_values.size());
+              cell_data->cells.emplace_back(dummy);
+              cell_data->reference_point_ptrs.emplace_back(
+                cell_data->reference_point_values.size());
             }
 
-          cell_data.reference_point_values.emplace_back(std::get<3>(i));
+          cell_data->reference_point_values.emplace_back(std::get<3>(i));
           send_permutation.emplace_back(std::get<5>(i));
         }
 
-      cell_data.reference_point_ptrs.emplace_back(
-        cell_data.reference_point_values.size());
+      cell_data->reference_point_ptrs.emplace_back(
+        cell_data->reference_point_values.size());
 
       this->ready_flag = true;
     }
 
 
 
+    template <int dim, int spacedim>
+    RemotePointEvaluation<dim, spacedim>::CellData::CellData(
+      const Triangulation<dim, spacedim> &triangulation)
+      : triangulation(triangulation)
+    {}
+
+
+
+    template <int dim, int spacedim>
+    std_cxx20::ranges::iota_view<unsigned int, unsigned int>
+    RemotePointEvaluation<dim, spacedim>::CellData::cell_indices() const
+    {
+      return {0, static_cast<unsigned int>(cells.size())};
+    }
+
+
+
+    template <int dim, int spacedim>
+    typename Triangulation<dim, spacedim>::active_cell_iterator
+    RemotePointEvaluation<dim, spacedim>::CellData::get_active_cell_iterator(
+      const unsigned int cell) const
+    {
+      AssertIndexRange(cell, cells.size());
+      return {&triangulation,
+              static_cast<int>(cells[cell].first),
+              static_cast<int>(cells[cell].second)};
```

Comment:
That's a bit nasty. I needed a cast to avoid a warning due to the interface of https://www.dealii.org/developer/doxygen/deal.II/classTriaActiveIterator.html#a999c78e642dec06e4f0ea8886524feb4. I am not sure if we want this.

Reply:
Okay, this would not the only place where the cast is necessary. I will further investigate this asap.